test/Path/Gen.hs

```diff
@@ -76,3 +76,6 @@ genFilePath = listOf genPathyChar
 
 genPathyChar :: Gen Char
 genPathyChar = frequency [(2, arbitrary), (1, elements "./\\")]
+
+shrinkPath :: Path a b -> [Path a b]
+shrinkPath (Path s) = Path <$> shrink s
```

Comment:
I'm not 100% sure but I think that the shrink function combined with a genValid generator should also only produce valid paths.

Also, would you mind configuring your editor or git to add newlines at the end? Otherwise we'll always have these noisy whitespace changes at the ends of files.